lib/scrabble.rb

```diff
@@ -15,4 +13,81 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+
+  def score(word)
+    if word == nil
+      return 0
+    elsif word.is_a? Integer
+      return 0
+    else
+      data = word.upcase.split("")
+      score_conversion_string(data)
+    end
+  end
+
+  def score_conversion_string(data)
+    if data == []
+      return 0
+    else
+      data.map do |letter|
+        @key[letter]
+      end.inject(:+)
+    end
+  end
+
+
+  def convert_letters_to_array(data)
+    data.map do |letter|
+      @key[letter]
+    end
+  end
+
+
+  def score_with_multipliers(data, multipliers, double = 1)
```

Comment:
This method is pretty lengthy. Looks like you were working on it up until the end of time, but I'd like to see you refactor out those nested blocks.